substrate/frame/revive/src/lib.rs

```diff
@@ -528,6 +538,63 @@ pub mod pallet {
 	#[pallet::storage]
 	pub(crate) type OriginalAccount<T: Config> = StorageMap<_, Identity, H160, AccountId32>;
 
+	/// The events emitted by this pallet while executing the current inflight transaction.
+	///
+	/// The events are needed to reconstruct the ReceiptInfo, as they represent the
+	/// logs emitted by the contract. The events are consumed when the transaction is
+	/// completed and moved to the `InflightTransactions` storage object.
+	#[pallet::storage]
+	#[pallet::unbounded]
+	pub(crate) type InflightEvents<T: Config> =
+		CountedStorageMap<_, Identity, u32, Event<T>, OptionQuery>;
```

Comment:
why did you choose to use StorageMap here vs StorageValue, if all these storage are killed at the end of the transaction, it's probably fine to use a StorageValue